gem/backends.py

```diff
@@ -36,15 +37,34 @@ def _update_user_from_claims(user, claims):
     user.email = claims.get("email", "")
     user.save()
 
+    username = claims.get("username", "")
+
     # If the user doesn't have a profile for some reason make one
     if not hasattr(user, 'profile'):
         user.profile = UserProfile(user=user)
+
+        # TODO: we should be using a more specific site here?
         user.profile.site = Site.objects.get(is_default_site=True)
 
     # Ensure the profile is linked to their auth service account using the uuid
     if user.profile.auth_service_uuid is None:
         user.profile.auth_service_uuid = claims.get("sub")
 
+        # If a user already exists with this username
+        # change that user's username
+        if username:
+            for u in User.objects.filter(
+                    username=username).exclude(pk=user.pk):
+                if user.profile.auth_service_uuid is None:
+                    user.username = str(user.profile.site.pk) + '_' + username
+                    user.save()
+                else:
+                    raise FieldError(
+                        'Desired username clashes with user with pk %s whose'
+                        ' profile has no auth_service_uuid' % u.pk)
```

Comment:
Typo? If we get here then the user that they are clashing with does have an `auth_service_uuid`